src/realm/alloc.hpp

```diff
@@ -52,8 +52,7 @@ int64_t to_int64(size_t value) noexcept;
 
 class MemRef {
 public:
-    MemRef() noexcept;
-    ~MemRef() noexcept;
+    MemRef() noexcept = default;
```

Comment:
FYI, this makes `const MemRef someVar;` ill-formed. I don't know if we care, but that is part of why I didn't make this change. If we did, `MemRef() noexcept {}` would allow that. (`=default` in the class body isn't a "user-defined" constructor)

`const auto someVar = MemRef();` works fine as-is and is IMO clearer anyway, FWIW.

Reply:
Why would that be ill-formed? It's certainly not very useful, but it compiles and I don't see any reason it'd be invalid...